Formula/sdl2.rb

```diff
@@ -41,7 +42,7 @@ def install
     # We have to do this because most build scripts assume that all SDL modules
     # are installed to the same prefix. Consequently SDL stuff cannot be
     # keg-only but I doubt that will be needed.
-    inreplace %w[sdl2.pc.in sdl2-config.in], "@prefix@", HOMEBREW_PREFIX
+    inreplace "sdl2.pc.in", "@prefix@", HOMEBREW_PREFIX
```

Comment:
I think we need to update this `inreplace` for `sdl2-config` too, since we don't want that to output a `Cellar` path.

Reply:
Patching `sdl2-config` is no longer necessary, in this release prefix is computed at runtime depending on `sdl2-config`'s location. Testing locally, I got the following results:
```
% sdl2-config --cflags --libs --version
-I/opt/homebrew/include/SDL2 -D_THREAD_SAFE
-L/opt/homebrew/lib -lSDL2
2.24.0
```

The `@prefix@` variable is no longer present in `sdl2-config.in` either, being replaced with this:
```sh
# Get the canonical path of the folder containing this script
bindir=$(cd -P -- "$(dirname -- "$0")" && printf '%s\n' "$(pwd -P)")

# Calculate the canonical path of the prefix, relative to the folder of this script
prefix=$(cd -P -- "$bindir/@bin_prefix_relpath@" && printf '%s\n' "$(pwd -P)")
```

Reply:
What happens if I call it as `/opt/homebrew/opt/sdl2/bin/sdl2-config --cflags --libs`?

Reply:
In this case it does point to the Cellar version:
```
% /opt/homebrew/opt/sdl2/bin/sdl2-config --cflags --libs
-I/opt/homebrew/Cellar/sdl2/2.24.0/include/SDL2 -D_THREAD_SAFE
-L/opt/homebrew/Cellar/sdl2/2.24.0/lib -lSDL2
```

Reply:
It sounds like we still need to patch it then. Maybe. What about
```
export PATH="/opt/homebrew/opt/sdl2/bin:$PATH"
sdl2-config --cflags --libs
```
?

Reply:
It still returns the cellar path unfortunately:
```
% export PATH="/opt/homebrew/opt/sdl2/bin:$PATH"
% sdl2-config --cflags --libs
-I/opt/homebrew/Cellar/sdl2/2.24.0/include/SDL2 -D_THREAD_SAFE
-L/opt/homebrew/Cellar/sdl2/2.24.0/lib -lSDL2
```

Reply:
Hmmm, okay. Let's leave this be for now, but we might need to patch it later on.

Reply:
The following patch restores the old behaviour:
```
diff --git a/Formula/sdl2.rb b/Formula/sdl2.rb
index fa8ff7a6050..d491355f38f 100644
--- a/Formula/sdl2.rb
+++ b/Formula/sdl2.rb
@@ -37,6 +37,7 @@ class Sdl2 < Formula
     # are installed to the same prefix. Consequently SDL stuff cannot be
     # keg-only but I doubt that will be needed.
     inreplace "sdl2.pc.in", "@prefix@", HOMEBREW_PREFIX
+    inreplace "sdl2-config.in", /bindir=.+\n/, "bindir=#{HOMEBREW_PREFIX}/bin"
 
     system "./autogen.sh" if build.head?
```

With it, `sdl2-config --cflags --libs` always returns:
```
-I/opt/homebrew/include/SDL2 -D_THREAD_SAFE
-L/opt/homebrew/lib -lSDL2
```

Would you like the patch to be applied before merging ?

Reply:
Thanks for working that out. I'm inclined to leave it be for now to see what it breaks (if anything).